python/taichi/shaders/Particles_vk.vert

```diff
@@ -42,4 +44,10 @@ void main() {
   } else {
     selected_color = in_color;
   }
+
+  if (ubo.has_per_vertex_radius == 0) {
+    selected_radius = vec2(ubo.radius, 0);
+  } else {
+    selected_radius = in_texcoord;
+  }
```

Comment:
Simply outputting `selected_radius` isn't enough. You will also need to actually use that radius (instead of the one in the uniform block) in the fragment shader. You'll also need to edit the .frag file for this.